pitop/robotics/navigation/navigation_controller.py

```diff
@@ -0,0 +1,323 @@
+from threading import Thread, Event
+import time
+from dataclasses import dataclass
+import numpy as np
+import math
+from typing import Union
+from simple_pid import PID
+import sched
+
+
+@dataclass
+class RobotState:
```

Comment:
A bit vaguely named

Reply:
I'm not sure how to make this more clear, it's the full state of the robot - this includes all degrees of freedom for this robot and their derivates. Any suggestions?